packages/keyring-controller/src/KeyringController.ts

```diff
@@ -573,17 +573,22 @@ export class KeyringController extends BaseController<
     if (!primaryKeyring) {
       throw new Error('No HD keyring found');
     }
-    const oldAccounts = await this.getAccounts();
+    const oldAccounts = await primaryKeyring.getAccounts();
```

Comment:
If we change this here, then we probably have to change what's being used by the extension as `accountCount`, as we currently count _all_ accounts there: https://github.com/MetaMask/metamask-extension/blob/6e0483da7d5269cbb3afdba2ea579902e9d2987b/app/scripts/metamask-controller.js#L3602

Reply:
for this reason we should mark this PR as breaking

Reply:
Good point, we should mark this as breaking and also make the change in the clients as well.